runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java

```diff
@@ -1132,22 +1162,7 @@ private void onConnectionData(
 
                 KafkaClientStream stream = streamsByInitialIds.get(initialId);
 
-                int newFlags = flags;
-
-                if (beforeResponseBytes == 0 && responseBytes > 0)
-                {
-                    newFlags = FLAG_INIT;
-                }
-                else if (beforeResponseBytes != 0 && responseBytes == 0)
-                {
-                    newFlags = FLAG_FIN;
-                }
-                else if (responseBytes > 0)
-                {
-                    newFlags = FLAG_NONE;
-                }
-
-                stream.doStreamData(traceId, newFlags, sequence, acknowledge, reserved,
+                stream.doStreamData(traceId, flags, sequence, acknowledge, reserved,
```

Comment:
I think the safest thing to do is always set `INIT` and `FIN` flags here, just to make sure we never send inconsistent flags to any stream sharing the connection.

```suggestion
                stream.doStreamData(traceId, flags | FLAG_INIT | FLAG_FIN, sequence, acknowledge, reserved,
```